### ISSUE TITLE
Weight decay handled differently in torch.optim.radam compared to original implementation

### ISSUE THREAD

**Issue by kevjshih** [pytorch/pytorch#88836]
### 🐛 Describe the bug

Comparing the implementation in torch.optim.radam with https://github.com/LiyuanLucasLiu/RAdam/blob/master/radam/radam.py , I noticed that the original author's implementation applies weight decay at the very end as a decoupled step, whereas the torch.optim.radam implementation applies weight decay immediately. The difference is significant because only in the latter does weight decay affect the exponential average computation. 

Also worth noting, the Radam paper specifically mentions using the decoupled weight norm formulation by citing the AdamW paper in the first paragraph of the experiments section.

### Versions

Based on the radam implementation in the current master branch.

cc @vincentqb @jbschlosser @albanD

**Comment by albanD** [comment]
Hi,

Thanks for the report!
We did follow the algorithm from the paper but it doesn't mention weight decay there. So we used the same approach as we usually do for weight decay here.

I'm afraid that changing this default would be BC-breaking so it will be quite hard for us to do. Maybe we want to introduce a RAdamW?

**Comment by albanD** [comment]
Sent a PR with the note: https://github.com/pytorch/pytorch/pull/89575
Let me know what you think!

**Comment by kevjshih** [comment]
Looks good to me!
